FE scale audit: Code graph uploads crash on large repos

As part of the [FE scale audit](https://docs.google.com/spreadsheets/d/1tqpi22csQJNlk7Q9JFWcFj09VbWOiFdp-Vf18qJsWHc/edit#gid=0), we noticed that code graph uploads inside the repo page are crashing and becoming unable to use on larger repos:

```
react_devtools_backend.js:4026 Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Yi (react-dom.production.min.js:264:247)
    at Qu (react-dom.production.min.js:179:340)
    at Object.queryConnection (CodeIntelUploadsPage.tsx:259:29)
    at c.project (FilteredConnection.tsx:270:30)
    at c._next (switchMap.ts:28:27)
    at c.next (Subscriber.ts:49:18)
    at s._tryNext (scan.ts:55:26)
```

This can be reproduced on dotcom, e.g. on the VS Code repo: https://sourcegraph.com/github.com/microsoft/vscode/-/code-graph/uploads

Here is a screenshot of the issue: https://www.figma.com/file/SiFt7KlbjUwSMXPfsXXdew/Strategic-readiness---design?node-id=135%3A3

**Note: The error seems similar to #43335, so maybe a it's the same fix?**

--- a/client/web/src/enterprise/codeintel/indexes/pages/CodeIntelIndexesPage.tsx
+++ b/client/web/src/enterprise/codeintel/indexes/pages/CodeIntelIndexesPage.tsx
@@ -130,6 +130,22 @@ export const CodeIntelIndexesPage: FunctionComponent<CodeIntelIndexesPageProps> 
 
     const deletes = useMemo(() => new Subject<undefined>(), [])
 
+    const queryConnection = useCallback(
+        (args: FilteredConnectionQueryArguments) => {
+            setArgs({
+                query: args.query ?? null,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+                state: (args as any).state ?? null,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+                isLatestForRepo: (args as any).isLatestForRepo ?? null,
+                repository: repo?.id ?? null,
+            })
+            setSelection(new Set())
+            return queryIndexes(args)
+        },
+        [queryIndexes, repo?.id]
+    )
+
     return (
         <div className="code-intel-indexes">
             <PageTitle title="Auto-indexing jobs" />
@@ -257,18 +273,7 @@ export const CodeIntelIndexesPage: FunctionComponent<CodeIntelIndexesPageProps> 
                         querySubject={querySubject}
                         nodeComponent={CodeIntelIndexNode}
                         nodeComponentProps={{ now, selection, onCheckboxToggle }}
-                        queryConnection={args => {
-                            setArgs({
-                                query: args.query ?? null,
-                                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-                                state: (args as any).state ?? null,
-                                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-                                isLatestForRepo: (args as any).isLatestForRepo ?? null,
-                                repository: repo?.id ?? null,
-                            })
-                            setSelection(new Set())
-                            return queryIndexes(args)
-                        }}
+                        queryConnection={queryConnection}
                         history={history}
                         location={location}
                         cursorPaging={true}

--- a/client/web/src/enterprise/codeintel/uploads/pages/CodeIntelUploadsPage.tsx
+++ b/client/web/src/enterprise/codeintel/uploads/pages/CodeIntelUploadsPage.tsx
@@ -161,6 +161,22 @@ export const CodeIntelUploadsPage: FunctionComponent<React.PropsWithChildren<Cod
 
     const deletes = useMemo(() => new Subject<undefined>(), [])
 
+    const queryConnection = useCallback(
+        (args: FilteredConnectionQueryArguments) => {
+            setArgs({
+                query: args.query ?? null,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+                state: (args as any).state ?? null,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+                isLatestForRepo: (args as any).isLatestForRepo ?? null,
+                repository: repo?.id ?? null,
+            })
+            setSelection(new Set())
+            return queryLsifUploads(args)
+        },
+        [queryLsifUploads, repo?.id]
+    )
+
     return (
         <div className="code-intel-uploads">
             <PageTitle title="Code graph data uploads" />
@@ -255,18 +271,7 @@ export const CodeIntelUploadsPage: FunctionComponent<React.PropsWithChildren<Cod
                         pluralNoun="uploads"
                         nodeComponent={CodeIntelUploadNode}
                         nodeComponentProps={{ now, selection, onCheckboxToggle }}
-                        queryConnection={args => {
-                            setArgs({
-                                query: args.query ?? null,
-                                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-                                state: (args as any).state ?? null,
-                                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-                                isLatestForRepo: (args as any).isLatestForRepo ?? null,
-                                repository: repo?.id ?? null,
-                            })
-                            setSelection(new Set())
-                            return queryLsifUploads(args)
-                        }}
+                        queryConnection={queryConnection}
                         history={history}
                         location={props.location}
                         cursorPaging={true}

